src/FlagProvider.tsx

```diff
@@ -1,70 +1,47 @@
-import * as React from 'react';
-import FlagContext from './FlagContext';
+import React, { useEffect, useRef } from 'react';
 import { UnleashClient, IConfig, IContext } from 'unleash-proxy-client';
+import FlagContext from './FlagContext';
 
 interface IFlagProvider {
   config: IConfig;
+  client?: UnleashClient;
 }
 
-const FlagProvider: React.FC<IFlagProvider> = ({ config, children }) => {
-  const [client, setClient] = React.useState(null);
-  const functionCalls = React.useRef<any>();
-  functionCalls.current = [];
-
-  React.useEffect(() => {
-    const client = new UnleashClient(config);
+const FlagProvider: React.FC<IFlagProvider> = (props) => {
+  const client = useRef<UnleashClient>(props.client);
 
-    functionCalls.current.forEach((call: any) => {
-      call(client);
-    }, []);
+  if (!client.current) {
+    client.current = new UnleashClient(props.config);
 
-    functionCalls.current = [];
-
-    setClient(client);
+    if (Array.isArray(props.config.bootstrap)) {
+      // @ts-ignore: Toogles should be set synchronously for proper SSR support
+      client.current.toggles = props.config.bootstrap;
```

Comment:
I do not understand why this is needed? Would not the constructor of the client handle the boostrap directly? Or it might be a issue that it also want to update the local storage, which might be async?